lib/internal/url.js

```diff
@@ -286,38 +283,43 @@ Object.defineProperties(URL.prototype, {
         ret += '//';
         const has_username = typeof ctx.username === 'string';
         const has_password = typeof ctx.password === 'string' &&
-                             ctx.password !== '';
-        if (has_username || has_password) {
+                              ctx.password !== '';
+        if (options.auth && (has_username || has_password)) {
           if (has_username)
             ret += ctx.username;
           if (has_password)
             ret += `:${ctx.password}`;
           ret += '@';
         }
-        if (unicode) {
-          ret += punycode.toUnicode(this.hostname);
-          if (this.port !== undefined)
-            ret += `:${this.port}`;
-        } else {
-          ret += this.host;
-        }
+        ret += options.unicode ?
+          domainToUnicode(this.host) : this.host;
```

Comment:
(Strange, my response ended up both in my own review and here, but when I deleted that one in my review this one disappeared too)..

I think `options` is neither in the spec nor in the old `url.format`. For me the meaning of `unicode` is pretty straightforword(maybe it's just me though). The two serialization alternatives are named `Domain to ASCII` and `Domain to Unicode` in the spec.